test/libsolidity/semanticTests/various/create_random.sol

```diff
@@ -0,0 +1,23 @@
+contract A {
+    uint256 public immutable createdOn = block.number;
+}
+
+contract C {
+    function addr() external returns (address) {
+        return address(this);
+    }
+
+    function test() external returns (address a1, address a2) {
+        bytes memory creation = type(A).creationCode;
+
+        assembly {
+            a1 := create(0, add(creation, 32), mload(creation))
+            a2 := create2(0, add(creation, 32), mload(creation), address())
+        }
+    }
+}
+
+// ----
+// addr() -> 0xbe9efd422ee479f010753a6bc06afe3a8444fc00
+// test() -> 0xf467baad569be981fb1a781f640de36059cd774a, 0x4e76c5b8db09c0f1be7f40921d7fec817e358ab7
```

Comment:
The values don't quite match
```
>>> cast compute-address 0xbe9efd422ee479f010753a6bc06afe3a8444fc00 --nonce 0
Computed Address: 0xC1397e3D7f722ba7F2BBA94054a8f2D3D66E296A
>>> cast compute-address 0xbe9efd422ee479f010753a6bc06afe3a8444fc00 --nonce 1
Computed Address: 0x640DE36059cD774a4fd957bf6Cf8EEF4491E431c
>>> cast compute-address 0xbe9efd422ee479f010753a6bc06afe3a8444fc00 --nonce 2
Computed Address: 0x5e01F63D4C8ea2060e389A2bC5c408CFCC02e421
```

Reply:
```
create from: 1212121212121212121212121212120000000012 nonce: 1 -> be9efd422ee479f010753a6bc06afe3a8444fc00
create from: be9efd422ee479f010753a6bc06afe3a8444fc00 nonce: 1 -> f467baad569be981fb1a781f640de36059cd774a
```

So it turns out the problem is the conversion from h256 -> h160 is truncated from the wrong side, left-aligned vs. right-aligned.